torch/csrc/autograd/custom_function.h

```diff
@@ -315,12 +315,10 @@ variable_list CppNode<T>::apply(variable_list&& inputs) {
 
   auto outputs = T::backward(&ctx_, backward_inputs);
 
-  // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
-  int num_forward_inputs = is_variable_input_.size();
-  auto num_outputs = outputs.size();
+  const auto num_forward_inputs = static_cast<int64_t>(is_variable_input_.size());
```

Comment:
Why can't these two stay size_t?

Reply:
`num_outputs` comes from `auto outputs = T::backward(&ctx_, backward_inputs);` where `T` is a template. I suspect the type of `outputs.size()` varies for different `T`.

Reply:
Ho interesting. In my mind we were enforcing that backward always returns a variable_list. But that is not the case indeed.